astropy/io/fits/card.py

```diff
@@ -1299,20 +1299,21 @@ def _format_value(value):
 
 def _format_float(value):
     """Format a floating number to make sure it gets the decimal point."""
-    value_str = f"{value:.16G}"
-    if "." not in value_str and "E" not in value_str:
-        value_str += ".0"
-    elif "E" in value_str:
+    value_str = str(value)
+
+    if "e" in value_str:
         # On some Windows builds of Python (and possibly other platforms?) the
         # exponent is zero-padded out to, it seems, three digits.  Normalize
         # the format to pad only to two digits.
-        significand, exponent = value_str.split("E")
+        significand, exponent = value_str.split("e")
         if exponent[0] in ("+", "-"):
             sign = exponent[0]
             exponent = exponent[1:]
         else:
             sign = ""
         value_str = f"{significand}E{sign}{int(exponent):02d}"
+    elif "." not in value_str:
+        value_str += ".0"
 
     # Limit the value string to at most 20 characters.
```

Comment:
This is not part of your PR, but what follows is just *wrong* - if a shorter representation is wanted, one should round, not truncate!  It is sad that this limit exists, since in 23 characters everything is representable.

Reply:
Indeed truncating also looks very weird to me, but I did not want to change the way that is handled since it hasn't caused issues in my experience. Maybe this should be explored in another PR.

Reply:
Yes we should explore that in another issue/PR. The tricky part is that the FITS Standard distinguished fixed and free formats:
- fixed-format is required for values of mandatory keywords and is recommended for values of all other keywords.
- fixed-format: a float number must be right-justified in Bytes 11 through 30.
- free-format: may occur anywhere within Bytes 11 through 80.